firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/common/BytesBackedNativeSessionFile.java

```diff
@@ -53,12 +56,24 @@ public InputStream getStream() {
     return isEmpty()
         ? null
         : CrashlyticsReport.FilesPayload.File.builder()
-            .setContents(bytes)
+            .setContents(asGzippedBytes())
             .setFilename(dataTransportFilename)
             .build();
   }
 
   private boolean isEmpty() {
     return bytes == null || bytes.length == 0;
   }
+
+  @Nullable
+  private byte[] asGzippedBytes() {
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
```

Comment:
Is there value in using the same `gzip` function in both classes, so as to be assured that they, in fact, work the same way?

Reply:
Talked about this offline. Decided it's fine.